File: sdk/src/main/java/me/digi/sdk/core/entities/DataRequest.java
/*
 * Copyright (c) 2009-2019 digi.me Limited. All rights reserved.
 */
package me.digi.sdk.core.entities;

import android.support.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

// Confluence ContractSchemav2-DataRequest
public class DataRequest implements Serializable {

    // Not yet supported
    /*@SerializedName("serviceGroups")
    public List<ServiceGroup> objectTypes;*/

    @SerializedName("timeRanges")
    public List<TimeRange> timeRanges;

    public DataRequest(@NonNull List<TimeRange> timeRanges) {
        this.timeRanges = timeRanges;
    }
}

File: sdk/src/main/java/me/digi/sdk/entities/DataRequest.kt
package me.digi.sdk.entities

interface DataRequest {
    var serviceGroups: List<ServiceGroup>
    var timeRanges: List<TimeRange>
    val context: String
    val criteria: List<Any>

    fun serviceGroupsInitialized(): Boolean
    fun timeRangesInitialized(): Boolean
    fun criteriaInitialized(): Boolean
}